src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Unit.Tests/CompilationHelper.cs

```diff
@@ -12,15 +12,37 @@
 using System.Text.Encodings.Web;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Text;
 using Xunit;
 
 namespace System.Text.Json.SourceGeneration.UnitTests
 {
+    public record JsonSourceGeneratorResult
+    {
+        public Compilation NewCompilation { get; set; }
+        public SourceGenerationSpec? SourceGenModel { get; set; }
+        public ImmutableArray<Diagnostic> Diagnostics { get; set; }
+
+        public IEnumerable<TypeGenerationSpec> AllGeneratedTypes
+            => SourceGenModel is { } model
+                ? model.ContextGenerationSpecs.SelectMany(ctx => ctx.GeneratedTypes)
+                : Array.Empty<TypeGenerationSpec>();
+
+        public void AssertContainsType(string fullyQualifiedName)
+            => Assert.Contains(
+                    AllGeneratedTypes,
+                    spec => spec.TypeRef.FullyQualifiedName == fullyQualifiedName);
+    }
+
     public class CompilationHelper
     {
         private static readonly CSharpParseOptions s_parseOptions =
             new CSharpParseOptions(kind: SourceCodeKind.Regular, documentationMode: DocumentationMode.Parse);
 
+#if ROSLYN4_0_OR_GREATER
```

Comment:
Does this PR provide an opportunity to switch to some helpers in `RoslynTestUtils`? https://github.com/dotnet/runtime/blob/main/src/libraries/Common/tests/SourceGenerators/RoslynTestUtils.cs